Fix text in docs (#6253)
reactjs/react.dev: src/content/learn/synchronizing-with-effects.md

Author: I think it's clearer in this way, maybe I'm wrong?

  }
```

> **This illustrates that if remounting breaks the logic of your application, this usually uncovers existing bugs.** From the user's perspective, visiting a page shouldn't be different from re-visiting it (clicking a link on the page, and pressing Back). React verifies that your components abide by this principle by remounting them once in development.

Reviewer: ```suggestion
**This illustrates that if remounting breaks the logic of your application, this usually uncovers existing bugs.** From a user's perspective, visiting a page shouldn't be different from visiting it, clicking a link, then pressing Back to view the page again. React verifies that your components abide by this principle by remounting them once in development.
```
